pkg/aws/cloud.go

```diff
@@ -91,3 +97,29 @@ func (c *defaultCloud) AccountID() string {
 func (c *defaultCloud) Region() string {
 	return c.cfg.Region
 }
+
+func (c *defaultCloud) Validate() (appMeshAccessError, cloudMapAccessError error) {
```

Comment:
I came across SimulatePrincipalPolicy API [1] that allows programatic way to validate and audit permissions for IAM users, groups, and roles. Should we use `SimulatePrincipalPolicy` instead with the list of `ActionNames` and checking their `EvalDecision`? 

[1] https://docs.aws.amazon.com/sdk-for-go/api/service/iam/#IAM.SimulatePrincipalPolicy